pkg/webui/console/components/pubsub-form/index.js

```diff
@@ -74,24 +74,28 @@ export default class PubsubForm extends Component {
 
     const { initialPubsubValue, update } = this.props
 
-    const initialIsMqtt = update && 'mqtt' in initialPubsubValue
-    const initialMqttSecure = initialIsMqtt ? initialPubsubValue.mqtt.use_tls : false
-    const initialUseCredentialsMqtt = initialIsMqtt
-      ? Boolean(initialPubsubValue.mqtt.username || initialPubsubValue.mqtt.password)
-      : true
-
-    const initialIsNats = update && 'nats' in initialPubsubValue
-    const { password, username } = initialIsNats
-      ? mapNatsServerUrlToFormValue(initialPubsubValue.nats.server_url)
-      : { password: undefined, username: undefined }
-    const initialUseCredentialsNats = initialIsNats ? Boolean(password || username) : true
-
     this.state = {
       error: '',
-      isMqtt: initialIsMqtt,
-      mqttSecure: initialMqttSecure,
-      mqttUseCredentials: initialUseCredentialsMqtt,
-      natsUseCredentials: initialUseCredentialsNats,
+      mqttUseCredentials: true,
```

Comment:
```suggestion
      provider: blankValues._provider
      mqttUseCredentials: true,
```

You don't account for non-update forms. Currently the pub/sub creation form does not render neither section.